Admin oauth: Option to create admin user in case it does not exist

imartinezcs opened the pull request:
Add the option to create an admin user in case if it doesn't exist, as long as the email domain matches the configured one. 

A new setting field (boolean) would be used for this: **ADMIN_OAUTH_CREATE_USER_IF_MISSING**.

This feature would be so usable in case you (or your development team/company) mantain several websites and want to grant admin access to everybody within that group without creating each user in each website.

matthiask reviewed (commented):
I think I like it. 

I think there should be a fallback if `ADMIN_OAUTH_CREATE_USER_IF_MISSING` isn't set. I also think it might be better to e.g. add a `create_user` argument to `admin_oauth` which receives the request and the email address as arguments; the method in turn can create the user with the rights you want. 

Next, `authenticate` could also fail because the user exists but is inactive. Your code would blindly try to create a new user and crash with an integrity error. You should probably try handling this in the proposed `create_user` callable somehow (I would be happy with providing a callable which people can use directly instead of having to redefine everything themselves, but it shouldn't be active by default)

Next, `auth.login` without a preceeding `auth.authenticate` call fails because the login function doesn't know the authentication backend if one isn't provided, or at least that was the case once upon a time. 

Please add some tests as well for this.

imartinezcs commented:
Hi Matthias, thank you for your advices. I've updated the repository taking into account your suggestions. Here are the main changes:
- New `ADMIN_OAUTH_CREATE_USER_CALLBACK` setting: it defines the user creation method path. Default value is `authlib.admin_oauth.views.create_admin_user`
- Related to this, I added the default method `create_admin_user`, which creates a user with staff role but no more permissions. It also checks if a inactive user with that email exists before creating it.
- New tests added in order to verify proper responses for different cases:
> Create user if not existing
> Do not create user if exists in database and is not active
> Response proper message if provided user creation method is `None`
> Response proper message if provided method path cannot be imported

imartinezcs commented:
Sorry, I closed the PR by mistake.

imartinezcs commented:
Hi Matthias, could you take a look in the proposed changes? Thank you in advance.

matthiask commented:
Thank you! I have applied a few changes and will push out a release shortly:

- I only use a single setting now.
- The code errors out if the callback cannot be imported instead of showing error messages to users. I don't like server crashes too much but I think they are better than the alternative of showing users errors they cannot do anything about.

imartinezcs commented:
Thank you!

imartinezcs commented:
Hi @matthiask I have installed the 0.15.0 version, but now I don't see the "Log in with Google" button in the admin login page. I've checked that v0.14.0 works properly with same settings. Could you take a look on that please?
